### PR TITLE
fix(plugin/azure-function): clear upstream uri and request uri inject plugin logic

### PR DESCRIPTION
**Pull request by oowl**
### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference
KAG-2841

### CODE REVIEW

**Comment by oowl** [comment]
@bungle Hi Apo, I do not know how this plugin is designed, but I think these upstream URIs and request URIs can not inject azure function plugin logic, Which do you mean?

File: changelog/unreleased/kong/fix-upstream-uri-azure-function-plugin.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+message: "**azure-functions**: clear upstream uri and request uri inject plugin logic"
```
Comment (tzssangglass) [inline changelog/unreleased/kong/fix-upstream-uri-azure-function-plugin.yml:1]: message is not user-facing enough

File: spec/03-plugins/35-azure-functions/01-access_spec.lua [hunk 2]
```diff
@@ -175,5 +205,18 @@ for _, strategy in helpers.each_strategy() do
...
+      local res = assert(proxy_client:send {
+        method  = "GET",
+        path    = "/",
+        query   = { hello = "world" },
+        headers = {
+          ["Host"] = "azure3.com"
+        }
+      })
+
+      assert(tonumber(res.headers["Content-Length"]) > 100)
```
Comment (tzssangglass) [inline spec/03-plugins/35-azure-functions/01-access_spec.lua:218]: Just a question, what does `res.headers["Content-Length") > 100` mean?

File: changelog/unreleased/kong/fix-upstream-uri-azure-function-plugin.yml (hunk 1 above)
Comment (windmgc) [inline changelog/unreleased/kong/fix-upstream-uri-azure-function-plugin.yml:1]:
```suggestion
message: "**azure-functions**: azure-functions plugin now eliminates upstream/request URI and only use `routeprefix` configuration field to construct request path when requesting Azure API"
```

**Review by windmgc — APPROVED** [review]
The change seems reasonable to me since concatenating the upstream URI/request URI with the `conf.routeprefix` is quite unnecessary, it seems we only need the `routeprefix` to construct the Azure API request
Approved although it might be a potential behavior change

File: spec/03-plugins/35-azure-functions/01-access_spec.lua (hunk 2 above)
Comment (oowl) [reply spec/03-plugins/35-azure-functions/01-access_spec.lua:218]: That's no other means, assert that an approved response from Mockbin is expected.

**Review by tzssangglass — APPROVED** [review]

**Comment by brentos** [comment]
@Tieske Do you happen to know / remember why this plugin concatenated the upstream path? This PR changes the existing behavior. It honestly seems like the common case would be to create a serviceless route.

**Comment by jschmid1** [comment]
@Tieske do you have any further information?

**Comment by bungle** [comment]
I don't recall anything about this either.

**Review by brentos — COMMENTED** [review]
I'm okay with the changes but I think we should make this a "breaking_change" in the changelog, since some users could depend on this behavior

**Comment by oowl** [comment]
@brentos I have changed the entry to breaking change type. Please help me review it again.

**Review by brentos — APPROVED** [review]

**Pull request merged**